staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/generic/matcher.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generic
+
+import (
+	"context"
+	celtypes "github.com/google/cel-go/common/types"
+	celconfig "k8s.io/apiserver/pkg/apis/cel"
+
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/admission/plugin/cel"
+	"k8s.io/klog/v2"
+)
+
+var _ cel.ExpressionAccessor = &cel.MatchCondition{}
+
+//// MatchCondition contains the inputs needed to compile, evaluate and validate a cel expression
+//type MatchCondition struct {
+//	Name       string
+//	Expression string
+//}
+//
+//func (v *MatchCondition) GetExpression() string {
+//	return v.Expression
+//}
+
+var _ Matcher = &matcher{}
+
+// matcher evaluates compiled cel expressions and determines if they match the given request or not
+type matcher struct {
+	filter cel.Filter
+}
+
+func NewMatcher(filter cel.Filter) Matcher {
+	return &matcher{
+		filter: filter,
+	}
+}
+
+// Match takes a list of Evaluation converts them into actionable a decision of whether they match or not.
+// If no match returns name of failed condition
+func (v *matcher) Match(ctx context.Context, versionedAttr *admission.VersionedAttributes) (bool, string, error) {
```

Comment:
Re-anchoring comment that this should take VersionedParams so it can be re-used for in-process enforcement

Reply:
done!